internal/config/adapter.go

```diff
@@ -27,3 +29,13 @@ func (a Adapters) GetMap() map[string]struct{} {
 
 	return res
 }
+
+func (a Adapter) IsTransactional() bool {
+	txMap := map[Storage]struct{}{
```

Comment:
it's better to move this map to `storage.go`
```suggestion
	var transactionalStorages map[Storage]bool
```

And then here we can do this:
```golang
    return transactionalStorages[a.Storage]
```